File: 프로그래머스/lv0/120851. 숨어있는 숫자의 덧셈 （1）/숨어있는 숫자의 덧셈 （1）.js
function solution(my_string) {
    var answer = 0;
    var arr = my_string.split("");
    //console.log(arr);
    
    for(var i=0; i<arr.length; i++){
        if(Number(arr[i]))
        answer += Number(arr[i]);
        console.log(answer);
    }
    return answer;
    
}


//다른사람 풀이
// isNaN 사용

//function solution(my_string) {
//    const arr = my_string.split('');
//    let sum = 0;
//    for (let i = 0; i < arr.length; i++) {
//        if (!isNaN(arr[i])) sum += Number(arr[i]);
//    }
//    return sum;
//}

File: 프로그래머스/lv0/120851. 숨어있는 숫자의 덧셈 （1）/숨어있는 숫자의 덧셈 （1）.java
class Solution {
    //정규식? 사용해서 숫자를 구분??
    // [^0-9] : 문자열 중에서 0~9를 제외한 문자열 (a~z, A~Z)
    // [0-9] : 문자열 중에서 0~9 까지 숫자가 들어 있는 문자열은 모두 해당
    public int solution(String my_string) {
        int answer = 0;
        String str = my_string.replaceAll("[^0-9]","");
        String[] arr = str.split("");
        for(int i=0; i<arr.length; i++){
            answer += Integer.parseInt(arr[i]);
        }
        return answer;
    }
}